fix(api): corriger le chemin dist et ajouter la dépendance contracts

## Problème

1. **Chemin dist incorrect** : `start:prod` dans `apps/api/package.json` pointait vers `dist/src/main.js` au lieu de `dist/main.js`, causant un crash au démarrage en production (`Cannot find module 'dist/main'`).

2. **Dépendance contracts manquante** : `@kraak/contracts` n'était pas déclaré dans les dépendances de l'API, causant des erreurs de résolution au build.

3. **Artefacts compilés hors dist/** : Des fichiers `.d.ts`, `.js` et `.js.map` compilés se retrouvaient à la racine de certains packages au lieu de rester dans `dist/`, polluant le dépôt.

## Correction

- `apps/api/package.json` : script `start:prod` corrigé → `node dist/main.js` ; ajout de `@kraak/contracts: "workspace:*"`
- `apps/api/tsconfig.build.json` : ajout de `rootDir: "./src"`, `paths: {}`, et `include: ["src"]` pour que `nest build` produise la bonne structure de sortie
- `.gitignore` : ajout de `*.d.ts` et `*.js.map` pour empêcher le suivi des artefacts compilés hors `dist/`
- Suppression de 15 fichiers d'artefacts stales

## Validation

- ✅ 374 tests unitaires / intégration passent (API Jest 13 + Web Vitest 33 + Mobile Vitest 21 + packages)
- ✅ Typecheck réussi pour web, mobile, et api
- ✅ Les 3 serveurs de dev démarrent correctement

--- a/.gitignore
+++ b/.gitignore
@@ -7,6 +7,10 @@ dist/
 .next/
 out/
 
+# Artefacts TypeScript compilés hors dist/
+*.d.ts
+*.js.map
+
 # Environnement
 .env
 .env.local

--- a/apps/api/package.json
+++ b/apps/api/package.json
@@ -9,7 +9,7 @@
     "build": "nest build",
     "start": "nest start",
     "start:dev": "nest start --watch",
-    "start:prod": "node dist/src/main.js",
+    "start:prod": "node dist/main.js",
     "lint": "eslint \"src/**/*.ts\"",
     "lint:fix": "eslint \"src/**/*.ts\" --fix",
     "test": "jest",
@@ -20,6 +20,7 @@
     "@nestjs/common": "^11.0.0",
     "@nestjs/core": "^11.0.0",
     "@nestjs/platform-express": "^11.0.0",
+    "@kraak/contracts": "workspace:*",
     "reflect-metadata": "^0.2.2",
     "rxjs": "^7.8.1"
   },

--- a/apps/api/tsconfig.build.json
+++ b/apps/api/tsconfig.build.json
@@ -1,4 +1,9 @@
 {
   "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "rootDir": "./src",
+    "paths": {}
+  },
+  "include": ["src"],
   "exclude": ["node_modules", "dist", "test", "**/*spec.ts"]
 }

